plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkContextFunction.java

```diff
@@ -61,11 +61,9 @@ public Object compute(IEclipseContext context, String contextKey) {
 
     Object path = context.get(PreferenceConstants.CLOUDSDK_PATH);
     File location = toFile(path);
-    CloudSdk.Builder builder = new CloudSdkProvider(null).createBuilder(location);
-    if (builder == null) {
-      return NOT_A_VALUE;
-    }
-    CloudSdk instance = builder.build();
+    // TODO(joaomartins): When can the SDK location in the context be different from the one in the
+    // preference store?
+    CloudSdk instance = new CloudSdkProvider().getCloudSdk();
```

Comment:
It would help me to understand why we shouldn't assume it. If here I should be getting the SDK location from the context instead of the Preference Store, I'd need to create a setter in CloudSdkProvider or add something to its constructor.